llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -17367,25 +17367,33 @@ SDValue PPCTargetLowering::LowerFRAMEADDR(SDValue Op,
   return FrameAddr;
 }
 
-// FIXME? Maybe this could be a TableGen attribute on some registers and
-// this table could be generated automatically from RegInfo.
-Register PPCTargetLowering::getRegisterByName(const char* RegName, LLT VT,
+#define GET_REGISTER_MATCHER
+#include "PPCGenAsmMatcher.inc"
```

Comment:
Most targets don't include AsmMatcher outside of their AsmPartner subtarget, see also https://reviews.llvm.org/D69130#inline-627969

It'd be nice if there was a way to have nicer deps here if possible.

(If this is the best that's possible, it's fine as-is. I thought I'd raise this though, and maybe there's a nice fix.)

Reply:
I agree it is a bit strange when I was doing it.  However the table gen function was exactly what I needed and I didn't want to duplicate it here.  Unfortunately I don't have a nice fix for this at this time.  Any suggestion?